### PR TITLE
Post on opam-install-dir

### PR DESCRIPTION
**Pull request by AltGr**

### CODE REVIEW

File: opam-install-dir.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+- `opam install --keep-build-dir` is now complemented with `--reuse-build-dir`,
+  for incremental builds within opam (assuming your build-system supports it
+  correctly)
+- `opam install --inplace-build` runs the scripts directly within the source
+  dir. If multiple packages are pinned to the same directory, they will be built
+  sequentially.
+- `opam install --working-dir` uses the working directory state of your project,
+  instead of the state registered in the version control system. Don't worry,
+  opam will warn you if you have uncommitted changes.
```
Comment (dbuenzli) [inline opam-install-dir.md]: It seems to me that if I request `--working-dir`, I don't want `opam` to warn me about uncommited changes.

File: opam-install-dir.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+### More related options
+
+A few other new options have been added to `opam install` and related commands,
+to improve the project-local workflows:
+
+- `opam install --keep-build-dir` is now complemented with `--reuse-build-dir`,
+  for incremental builds within opam (assuming your build-system supports it
+  correctly)
+- `opam install --inplace-build` runs the scripts directly within the source
```
Comment (dbuenzli) [inline opam-install-dir.md]: I think this should be `--in-place-build`

File: opam-install-dir.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+If you would really like automatic initialisation, and have a better interface
+to propose, your feedback is welcome!
+
+### More related options
+
+A few other new options have been added to `opam install` and related commands,
+to improve the project-local workflows:
+
+- `opam install --keep-build-dir` is now complemented with `--reuse-build-dir`,
```
Comment (dbuenzli) [inline opam-install-dir.md]: This is a bit unclear. Do I have to use both options to get the behaviour ? And is the option "sticky", or do I have to respecify it each time ?

**Comment by dbuenzli** [comment]
Thanks for redesigning all this, this all looks much better than `opam build` and is quite easy to understand which hints that the design is on a good track.

File: opam-install-dir.md (hunk 1 above)
Comment (AltGr) [inline opam-install-dir.md]: Ah, yes, that's the case, bad wording on my part!

**Comment by AltGr** [comment]
Thanks for reviewing, and for the feedback on `opam build` !

File: opam-install-dir.md (hunk 1 above)
Comment (Drup) [inline opam-install-dir.md]: I still really would like to give that option to `opam pin` and have opam remember it ...

**Comment by Drup** [comment]
If I do `opam install foo` and foo is both a (sub)directory and a package, does opam yell at me?

**Comment by AltGr** [comment]
All directories need to be explicit, e.g. `./foo`, otherwise opam assumes a package name (and yells if none if found). This sounded safer to me.

**Pull request merged**